### ISSUE TITLE
File-download component fails to link relative files

### ISSUE THREAD

**Issue by axonasif** [retypeapp/retype#69]
Hi, this is a bug report.
Assume my site is `example.com` with a file hierarchy of:

```
docs
|-- hello.md
|-- Music-One.mp3

README.md
```
Now if I use the following markdown code in `docs/hello.md`:

```markdown
[!file Music](Music-One.mp3)
```

It will result in `example.com/docs/hello/music-one.mp3` for the html output.
While it should have been `example.com/docs/Music-One.mp3` as `Music-One.mp3` file resides under `docs` dir and not `docs/hello` non-existent dir.

Also notice that the file name `Music-One.mp3` got lowercase and did not remain case-sensitive for the HTML output, thus resulting in unexpected behavior.

I would also suggest to use the HTML [download](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download) tag for prompting browsers to perform download for unrecognized downloadable file types.
Here is an example for such a type: https://raw.githubusercontent.com/supremegamers/wiki/main/common-issues/DarkMatter-Exo4.7.cfg

Thanks.

**Comment by geoffreymcgill** [comment]
Hi. Thanks for the report. I'm setting up a local test and we will figure out why this is failing.

**Comment by geoffreymcgill** [comment]
I reproduced and realized what's going on here. The reason `.mp3` links were not resolved correctly is related to the [`include`](https://retype.com/configuration/project/#include) project config.

There is a temp work-around. Add the following config to your projects `retype.json` file:

```json
{
  "include": [
    "*.mp3"
  ]
}
```

Once the `*.mp3` is added to the `include` config list, your sample and `[!file Music](Music-One.mp3)` will correctly resolve. 

Here's the explanation of why it does not resolve without the `*.mp3` config above.

By default, Retype will only resolve and copy over files from the following list:

```
*.gif
*.heif
*.jpeg
*.jpg
*.png
*.svg
*.webp
*.ai
*.bmp
*.eps
*.pdf
*.tiff
*.txt
*.zip
```

Or, files that are stored in the following folders:

```
**/static/**
**/public/**
**/assets/**
**/resources/**
```

This is basically a security feature as we wanted to help prevent authors from accidentally exposing content into the public. We've been slowly adding to the list of files type to automatically include and looks like we need to add `.mp3` and `.mp4` files to the list.

Another work-around to get your sample working would be to move your `Music-One.mp3` file into a folder with the folder name of `static`, `public`, `assets`, or `resources`

I'm going to create and internal issue for us to completely review this `known file type include` policy. I have some ideas on how we might be able to better handle this scenario.

Thanks again for reporting this issue. We will get this working smoother.

**Comment by geoffreymcgill** [comment]
This issue has now been fixed and will be included in the upcoming Retype `v1.8.0` release. 🎉 

We significantly improved the logic for handling how static resource files are copied over to the finished website, where now if you create a link to any local static file, that file will be automatically copied into the correct position in the `output`.

For instance, just creating the link to the `[Music](Music-One.mp3)` link would now be enough for Retype to recognize that the `.mp3` file is important and is to be copied to the finished website.

The `include` work-around mentioned [above](#issuecomment-877774952) will no longer be required.

The same logic is applied to all file types. Make a link to a local static file and that file will be included in the finished website build, which of course is how you would expect it to work. 😸
